### DISCUSSION TITLE
[Feature] ElasticSearch SQL 6.4 Support

### DISCUSSION THREAD

**Discussion by MrSantamaria** [grafana/grafana#56257] [Observability]
Available in beta ElasticSearch is introducing a SQL interpretation that will be more user friendly and highly configurable as oppose to lucene query.  
https://www.elastic.co/products/stack/elasticsearch-sql

Main features I'm interested in:
Custom queries not limited by lucene. 
Aggregation queries.
Changing legend names with the "as" keyword from SQL. 

Technically these should already work if a user is using ES 6.4+ but grafana would need to allow for the query editor feature on ES datasource.

**Comment by i5okie** [comment]
any progress here?

**Comment by guhcampos** [comment]
Allow me to register another request here.

One of the only drawbacks of using Elasticsearch pretty much anywhere is convincing people to learn Lucene (in the case of Grafana) or Elastic's Query DLS otherwise. Their implementation of SQL was a great improvement, but Grafana is falling behind in such regard. ES is a major platform even for timeseries data these days, so having SQL support is a big improvement.

I understand this probably means a major rewrite of the plugin, probably from scratch, but I believe it's worth considering. 

Thanks, for the great software!

**Comment by ghost** [comment]
Hello, any progress here please? 
Elasticsearch SQL is stable feature and its implementation for Elasticsearch datasource would be very helpful.

**Comment by pencilcap123** [comment]
any progress here ?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age of this issue, I am closing it. If support for ES SQL 6.4 is still needed, please reply here or feel free to make an [issue](https://github.com/grafana/grafana/issues) and tag it as a feature request. Thank you!
